src/protocol/AProtocol.php

```diff
@@ -71,7 +71,7 @@ protected function read(?int &$signature)
         $msg = '';
         while (true) {
             $header = $this->connection->read(2);
-            if (ord($header[0]) == 0x00 && ord($header[1]) == 0x00)
+            if ($msg !== '' && ord($header[0]) == 0x00 && ord($header[1]) == 0x00)
```

Comment:
I don't think it can happen. The origin of the bug I referenced was here. Somehow the server sent the chunk mark but didn't pass anything else. This signalled there is another chunk coming

Reply:
It seem you can be right. Because even bolt messages without response returns status response at least. Handshake has no server response at all.

Reply:
I've run it against both of our packages' tests and nothing crashed. My only fear is an infinite loop